### DISCUSSION TITLE
Where is the user being created in strapi?

### DISCUSSION THREAD

**Discussion by IamJoshing** [ghoshnirmalya/nextjs-strapi-boilerplate#149] [General]
I am looking to be pointed towards where the user is being created so I can update it to use the email as the username.

I have been playing with the boilerplate and different emails. I have different emails with the same username. When I try to login with a gmail account that has the same first name already set as an existing account the auth fails.

**Comment by ghoshnirmalya** [comment]
[Here](https://github.com/ghoshnirmalya/nextjs-strapi-boilerplate/blob/master/frontend/pages/api/auth/%5B...nextauth%5D.ts#L32-L34).

According to [Strapi documentation](https://strapi.io/documentation/developer-docs/latest/plugins/users-permissions.html#providers):
> Create a frontend route like FRONTEND_URL/connect/${provider}/redirect that have to handle the access_token param and that have to request STRAPI_BACKEND_URL/auth/${provider}/callback with the access_token param.
The JSON request response will be { "jwt": "...", "user": {...} }.

So, to comply with the above requirement, we do that [here](https://github.com/ghoshnirmalya/nextjs-strapi-boilerplate/blob/master/frontend/pages/api/auth/%5B...nextauth%5D.ts#L32-L34):

```ts
const response = await fetch(
  `${process.env.NEXT_PUBLIC_API_URL}/auth/${account.provider}/callback?access_token=${account?.accessToken}`
);
```

In the above case, **account.provider** can be **google** and **account.accessToken** will contain the **accessToken** returned by the NextAuth's [jwt callback](https://github.com/ghoshnirmalya/nextjs-strapi-boilerplate/blob/master/frontend/pages/api/auth/%5B...nextauth%5D.ts#L28) ([Link to documentation](https://next-auth.js.org/configuration/callbacks#jwt-callback)).

**Reply by IamJoshing** [reply]
Thanks! 

I got this to work by adding backend/app/extensions/users-permissions/services/Providers.js coping content from https://raw.githubusercontent.com/strapi/strapi/master/packages/strapi-plugin-users-permissions/services/Providers.js and changing the providers to use `const username = body.email;`

I am not sure if this is the strapi way or what other issues there may be.
